### PR TITLE
tide, Fix require_manually_triggered_jobs tide handling

### PR DESCRIPTION
**Pull request by oshoval**
When `require_manually_triggered_jobs` is used, tide doesn't wait for the jobs
that require manual trigger, nor adds them to batches.
This PR fix it, by checking the bp settings, and if the `require_manually_triggered_jobs` feature is enabled
for the branch, then force the jobs that have the following settings:
* `always_run=false`
* `optional=false`
* `skip_report=false`
* `run_if_changed` and `skip_if_only_changed` are empty

or has `run_before_merge=true` (as was already before this PR).

Fixes: https://github.com/kubernetes/test-infra/issues/31686
Fixes: https://github.com/kubernetes/test-infra/issues/31050

### CODE REVIEW

**Comment by oshoval** [comment]
Hi @jmguzik @petr-muller 
Can you please suggest if the direction is fine ? 
I will add unit tests of course once we see it is on the right track
Will update PR desc soon

Thanks

**Comment by petr-muller** [comment]
/ok-to-test

I think the idea is sound, happy to review when out of WIP and polished

**Comment by oshoval** [comment]
Thanks
Added unit tests
PR ready for review

File: prow/tide/tide_test.go
Comment (jmguzik) [inline prow/tide/tide_test.go:3046]: Excluding last two examples most tests have english-like descriptions like "doesn't run against branch" or "no matching presubmits (check cache eviction)", so I would change your proposition to match english-like style that was presented previously.

File: prow/tide/tide.go
Comment (jmguzik) [inline prow/tide/tide.go:1575]: small NIT, I would rephrase the name to something like `forceRun` etc to be consistent with other names.

**Review by jmguzik — COMMENTED** [review]
Many thanks for handling this edge-case. I added some comments.

**Comment by oshoval** [comment]
Thanks
Addressed comments

**Review by jmguzik — APPROVED** [review]
From my POV
/lgtm
/assign @petr-muller 
Let's wait on @petr-muller approval.

**Comment by oshoval** [comment]
Thanks

pull-test-infra-unit-test-race-detector-nonblocking isn't blocking iiuc, and also not due to this PR

For visibility, there is the function `GetBranchProtection`, but in case the Org is not found, it doesn't fall back to the global BP settings, this why i didn't use it.

Btw not part of this PR scope, but aint it redundant that `GetTideContextPolicy` and `presubmitsByPull` have lots in common but still different implementation even on the common parts, and also their result is populated into different elements ?
(I am totally not into every detail, so of course i might miss the point)

**Comment by oshoval** [comment]
oops commit had the word WIP, removed it
btw aint it should be blocked if there is a word WIP on commit title ?

**Review by petr-muller — APPROVED** [review]
Thanks!

**Comment by petr-muller** [comment]
> btw aint it should be blocked if there is a word WIP on commit title ?

I think the plugin only checks PR title, not commits

**Pull request merged**
